packages/miniflare/src/plugins/kv/assets.ts

```diff
@@ -0,0 +1,102 @@
+import path from "path";
+import { KVOptionsSchema } from "miniflare";
+import SCRIPT_KV_ASSETS from "worker:kv/assets";
+import { z } from "zod";
+import { Service, Worker_Binding } from "../../runtime";
+import { getAssetsBindingsNames, SharedBindings } from "../../workers";
+import { kProxyNodeBinding } from "../shared";
+import { KV_PLUGIN_NAME } from "./constants";
+
+export interface AssetsOptions {
+	assetsPath: string;
+	assetsKVBindingName?: string;
+	assetsManifestBindingName?: string;
+}
+
+export function isWorkersWithAssets(
+	options: z.infer<typeof KVOptionsSchema>
+): options is AssetsOptions {
+	return options.assetsPath !== undefined;
+}
+
+const SERVICE_NAMESPACE_ASSET = `${KV_PLUGIN_NAME}:asset`;
+
+export function buildAssetsManifest(): Uint8Array {
+	const buffer = new ArrayBuffer(20);
+	const assetManifest = new Uint8Array(buffer); // [0, 0, 0, ..., 0, 0]
+	assetManifest.set([1], 0); // [1, 0, 0, ..., 0, 0]
+
+	return assetManifest;
+}
+
+export async function getAssetsBindings(
+	options: AssetsOptions
+): Promise<Worker_Binding[]> {
+	const assetsBindings = getAssetsBindingsNames(
+		options?.assetsKVBindingName,
+		options?.assetsManifestBindingName
+	);
+
+	const assetsManifest = buildAssetsManifest();
+
+	return [
+		{
+			// this is the binding to the KV namespace that the assets are in.
+			name: assetsBindings.ASSETS_KV_NAMESPACE,
+			kvNamespace: { name: SERVICE_NAMESPACE_ASSET },
+		},
+		{
+			// this is the binding to an ArrayBuffer containing the binary-encoded
+			// assets manifest.
+			name: assetsBindings.ASSETS_MANIFEST,
+			data: assetsManifest,
+		},
+	];
+}
+
+export async function getAssetsNodeBindings(
+	options: AssetsOptions
+): Promise<Record<string, unknown>> {
+	const assetsManifest = await buildAssetsManifest();
+	const assetsBindings = getAssetsBindingsNames(
+		options?.assetsKVBindingName,
+		options?.assetsManifestBindingName
+	);
+
+	return {
+		[assetsBindings.ASSETS_KV_NAMESPACE]: kProxyNodeBinding,
+		[assetsBindings.ASSETS_MANIFEST]: assetsManifest,
+	};
+}
+
+export function getAssetsServices(options: AssetsOptions): Service[] {
+	// Use unsanitised file storage to ensure file names containing e.g. dots
+	// resolve correctly.
+	const persist = path.resolve(options.assetsPath);
```

Comment:
Yeah if this is already absolute then this can be removed.
The reason it would make it worse on Windows is that a Windows absolute path will look like:

```
C:\users\carmen\projects\foo
```

and resolving that with the posix version will do something like:

```
/current/working/directory/C:\users\carmen\projects\foo
```

😭

Reply:
It only makes sense to use `posix.resolve()` on a relative path.